File: app/local_plugins/ast_volcengine/android/src/main/kotlin/com/aiagent/ast_volcengine/AstVolcenginePlugin.kt
package com.aiagent.ast_volcengine

import com.aiagent.plugin_interface.NativeServiceRegistry
import io.flutter.embedding.engine.plugins.FlutterPlugin

class AstVolcenginePlugin : FlutterPlugin {

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        val context = binding.applicationContext
        // 火山引擎 AST 服务，注册多个 vendor 名（用户配置可能使用不同名称）
        val vendors = listOf("volcengine", "doubao", "bytedance")
        for (vendor in vendors) {
            NativeServiceRegistry.registerAst(vendor) { AstVolcengineService(context) }
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: app/local_plugins/vendors/ast_volcengine/ios/Classes/AstVolcenginePlugin.swift
import Flutter
import ai_plugin_interface
import os.log

/// Volcengine speech translation (AST) Flutter plugin entry.
///
/// Mirrors the Android `AstVolcenginePlugin`: registers vendors "volcengine",
/// "doubao" and "bytedance" against `AstVolcengineService`.
public class AstVolcenginePlugin: NSObject, FlutterPlugin {
    private static let log = OSLog(subsystem: "com.aiagent.ast_volcengine", category: "Plugin")

    public static func register(with registrar: FlutterPluginRegistrar) {
        let vendors = ["volcengine", "doubao", "bytedance"]
        for vendor in vendors {
            NativeServiceRegistry.shared.registerAst(vendor) { AstVolcengineService() }
        }
        os_log("Registered NativeAstService for vendors: %{public}@",
               log: log, type: .debug, vendors.description)
    }
}
